projects/dsp-ui/src/lib/search/search-panel/search-panel.component.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { OverlayModule } from '@angular/cdk/overlay';
+import { Component, ViewChild } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { MatMenuModule } from '@angular/material/menu';
+import { SearchPanelComponent } from './search-panel.component';
+
+/**
+ * Test host component to simulate parent component with a search panel.
+ */
+@Component({
+    template: `
+        <dsp-search-panel #searchPanelView
+            [route]="route"
+            [projectfilter]="projectfilter"
+            [expert]="expert"
+            [advanced]="advanced">
+        </dsp-search-panel>`
+})
+class TestHostComponent {
+
+    @ViewChild('searchPanelView') searchPanelComponent: SearchPanelComponent;
+
+    route = '/search';
+    projectfilter = true;
+    advanced = false;
+    expert = false;
+
+}
+
+describe('SearchPanelComponent', () => {
+    let testHostComponent: TestHostComponent;
+    let testHostFixture: ComponentFixture<TestHostComponent>;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                SearchPanelComponent,
+                TestHostComponent
+            ],
+            imports: [
+                OverlayModule,
+                MatMenuModule
+            ]
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(() => {
+        testHostFixture = TestBed.createComponent(TestHostComponent);
+        testHostComponent = testHostFixture.componentInstance;
+        testHostFixture.detectChanges();
+
+        expect(testHostComponent).toBeTruthy();
+    });
+
+    it('should create an instance', () => {
```

Comment:
@tobiasschweizer I do not know which other tests I have to implement for this component because it is only a "combiner" (component to assemble the search bar with other components). And it uses mostly CDK OverlayRef which I do not know how to define or mock in a unit test....

Reply:
you can always mock the children in your spec file. The just need to have the same selector and the same `@input`s, `@Output`s, and public methods. However, you do not really have to implement them. There are like a mock.

Reply:
done in 02d9312